### DISCUSSION TITLE
Welcome to bruh Discussions!

### DISCUSSION THREAD

**Discussion by wlib** [Technical-Source/bruh#1] [Announcements]
## 👋 Hey!

Bruh is very simple, but not everything is in the documentation.
If you are interested, want clarifications, or run into problems - this is the place to casually discuss.

Also, we are very welcome to new ideas or anything else relating to the project.
Don't hesitate to join or start a discussion!

**Comment by rodrigomv29** [comment]
Hey!  I just want clarification on why the name bruh? Is this in reference to the "bruh" meme?

**Reply by wlib** [reply]
Sorry for the late reply, I missed the notification.

It's not a meme, but the fact that people recognize it that way is good IMO. It's short, memorable, and not too serious.
I looked through npm's available package names when I was brainstorming the name, and "bruh" was one of (if not the) first name that I searched. I didn't expect it to be available, [and it wasn't](https://web.archive.org/web/20210106142558/https://npmjs.com/package/bruh). It had been abandoned for years with only [a default `package.json`](https://unpkg.com/browse/bruh@0.0.1/package.json). I assumed that this was a case of unintentional name squatting because the guy who pushed it was @kenwheeler (known for making slick-carousel). He probably was testing out `npm init`, typed in "bruh" for the name, and published it without realizing it. As far as naming goes, great minds think alike (but so do stupid ones lol).

Here's how I got the name:

> Daniel Ethridge
> Feb 17, 2021, 11:51 PM UTC
>
> The package name "bruh" has the default package.json with no other files, and is on version 0.0.1 since first publish 4 years ago. The owner does not have an accompanying repository on github (or anywhere listed - nothing is listed - everything is default) either. I assume that this publishing has caused a completely unintentional case of squatting, where the owner likely was experimenting with npm in the cli and accidentally actually pushed the package and then forgot about it immediately. I would like to have this package name for a (currently small) utilities library/web framework that I am actively developing for my own business, that will be open source. I read the "disputes" page which says that if the package is obviously squatted then ownership will be transferred without hassle, and I do not see how there could be a dispute here in what is very clearly an accidental package publish. Thank you for your help in advance!

> GitHub (GitHub Support)
> Feb 18, 2021, 7:18 PM UTC
>
> Hello,
>
> Thanks for reaching out.
>
> I can help with that. The bruh package has been transferred to the wlib user account and is all yours now, but you won’t be able to re-use any version numbers used by the previous author. In fact, I'd suggest publishing your first update as a major release, i.e., v1.0.0. Please publish an update to the package as soon as possible with your code.
>
> Please let us know if there is anything else we can do for you. We’ll be here to help.
>
> Thanks so much.
>
> Regards,
> Lisa

That's also the story of why bruh is at v1 but not considered stable just yet. I plan on making v2 a rewrite with typescript and a full test suite, and that will sort of be the real bruh v1.
